src/ziggurat/kafka_consumer/consumer.clj

```diff
@@ -0,0 +1,28 @@
+(ns ziggurat.kafka-consumer.consumer
+  (:require [clojure.tools.logging :as log]
+            [ziggurat.kafka-consumer.consumer-handler :refer :all])
+  (:import (java.util Map Properties)
+           (org.apache.kafka.clients.consumer KafkaConsumer ConsumerConfig)
+           (java.util.regex Pattern)))
+
+(defn- build-consumer-properties-map
+  [consumer-group-config]
+  (doto (Properties.)
+    (.putAll {ConsumerConfig/BOOTSTRAP_SERVERS_CONFIG        (:bootstrap-servers consumer-group-config)
+              ConsumerConfig/GROUP_ID_CONFIG                 (:consumer-group-id consumer-group-config)
+              ConsumerConfig/MAX_POLL_RECORDS_CONFIG         (int (or (:max-poll-records  consumer-group-config) 50))
+              ConsumerConfig/SESSION_TIMEOUT_MS_CONFIG       (int (or (:session-timeout-ms-config  consumer-group-config) 60000))
```

Comment:
Why are we not using Kafka consumer default values here? How did we decide on the current values as the default?

Reply:
Yes, we've changed max.poll.records to 500 which is the  default value. But, session.timeout.ms has been increased to 60 seconds because batch processing takes time and the processing time should fall under session.timeout.ms otherwise re-balancing happens.